src/python/pants/backend/go/subsystems/gotest.py

```diff
@@ -68,6 +69,19 @@ class GoTestSubsystem(Subsystem):
 
     skip = SkipOption("test")
 
+    with_race = BoolOption(
```

Comment:
How about `race_detector`?

Reply:
But why have this option? Users can use `__default__` to set the field on many targets.

Reply:
Because a user may want to run a test temporarily with the race detector enabled, just like they can do with `go test -race ...`. At least that's the thought.

Reply:
To clarify, the option is not intended to foreclose the user enabling the race detector by explicitly using the `test_with_race` field (or whatever the name becomes). Rather, the option is used in test contexts only when the field has not been specified on the `go_package`.